### PR TITLE
feat: Add helper methods to determine interactions integrations

### PR DESCRIPTION
**Pull request by DA-344**
## Summary

Adds ``Interaction.is_guild_integration`` and ``Interaction.is_user_integration`` to easily check whether an interaction has been invoked by a guild or user integration.

Also adds this to ``ApplicationContext``.

## Information

- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed).
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting,
      examples, ...).

## Checklist

- [x] I have searched the open pull requests for duplicates.
- [x] If code changes were made then they have been tested.
  - [x] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why.
- [ ] I have updated the changelog to include these changes.

### CODE REVIEW

**Review by Lulalaby — DISMISSED** [review]
Good addition! 💕🎉

**Review by Paillat-dev — APPROVED** [review]
👍

File: discord/commands/context.py [hunk 1]
```diff
@@ -345,6 +345,22 @@ def cog(self) -> Cog | None:
...
+        return self.interaction.is_guild_integration()
+
+    def is_user_integration(self) -> bool:
+        """:class:`bool`: Returns ``True`` if the invoked command was user installed.
+        This is a shortcut for :meth:`Interaction.is_user_integration`.
+
+        .. versionadded:: 2.7
+        """
+        return self.interaction.is_user_integration()
+
```
Comment (JustaSqu1d) [inline discord/commands/context.py:363]:
```suggestion
    def is_guild_integration(self) -> bool:
        """:class:`bool`: Checks if the invoked command is guild-installed.
        This is a shortcut for :meth:`Interaction.is_guild_integration`.

        .. versionadded:: 2.7
        """
        return self.interaction.is_guild_integration()

    def is_user_integration(self) -> bool:
        """:class:`bool`: Checks if the invoked command is user-installed.
        This is a shortcut for :meth:`Interaction.is_user_integration`.

        .. versionadded:: 2.7
        """
        return self.interaction.is_user_integration()

```

File: discord/interactions.py [hunk 2]
```diff
@@ -360,6 +360,26 @@ def followup(self) -> Webhook:
...
+        """:class:`bool`: Returns ``True`` if the interaction is a user integration.
+
+        .. versionadded:: 2.7
+        """
+        if self.user:
+            return self.authorizing_integration_owners.user_id == self.user.id
+
+        # This return should not be called but making sure it returns the expected value
+        return False
+
```
Comment (JustaSqu1d) [inline discord/interactions.py:382]:
```suggestion
    def is_guild_integration(self) -> bool:
        """:class:`bool`: Checks if the interaction is a guild integration.

        .. versionadded:: 2.7
        """
        if self.guild_id:
            return self.authorizing_integration_owners.guild_id == self.guild_id
        return False

    def is_user_integration(self) -> bool:
        """:class:`bool`: Checks if the interaction is a user integration.

        .. versionadded:: 2.7
        """
        if self.user:
            return self.authorizing_integration_owners.user_id == self.user.id

        # This return should not be called but to make sure it returns the expected value
        return False

```

File: CHANGELOG.md [hunk 3]
```diff
@@ -36,6 +36,8 @@ These changes are available on the `master` branch, but have not yet been releas
   `Permissions.use_external_sounds` and
   `Permissions.view_creator_monetization_analytics`.
   ([#2620](https://github.com/Pycord-Development/pycord/pull/2620))
+- Added helper methods to determine an `Interaction` integration type.
```
Comment (JustaSqu1d) [inline CHANGELOG.md:39]:
```suggestion
- Added helper methods to determine an `Interaction`'s integration type.
```

**Review by JustaSqu1d — CHANGES_REQUESTED** [review]
minor convention and grammar

**Review by JustaSqu1d — CHANGES_REQUESTED** [review]
I don't quite like the naming. Interactions aren't exactly integrations.

Could we change it from `is_x_integration` to `is_x_authorized`?

**Comment by DA-344** [comment]
> I don't quite like the naming. Interactions aren't exactly integrations.

Yeah, but when invoking an application command, you are using an integration, which can either be guild or user installed.

> Could we change it from `is_x_integration` to `is_x_authorized`?
>
> Or if you have a better name that we could go with.

I think the current naming is nice, but if it is _really_ neccesary I can change it to ``is_x_authorized``.

**Comment by Paillat-dev** [comment]
I feel like both namings make sense, maybe `is_x_authorized` is more explicit since that *is* in fact what is being checked.

**Comment by DA-344** [comment]
Makes sense, I have renamed the methods to ``is_x_authorised``, with its respective en-US aliases.

File: discord/commands/context.py [hunk 4]
```diff
@@ -345,6 +345,36 @@ def cog(self) -> Cog | None:
 
         return self.command.cog
 
+    def is_guild_authorised(self) -> bool:
+        """:class:`bool`: Checks if the invoked command is guild-installed.
+        This is a shortcut for :meth:`Interaction.is_guild_authorised`.
+
```
Comment (JustaSqu1d) [inline discord/commands/context.py:351]: Missing "There is an alias" note.

File: discord/commands/context.py [hunk 5]
```diff
@@ -345,6 +345,36 @@ def cog(self) -> Cog | None:
...
+        .. versionadded:: 2.7
+        """
+        return self.interaction.is_guild_authorised()
+
+    def is_user_authorised(self) -> bool:
+        """:class:`bool`: Checks if the invoked command is user-installed.
+        This is a shortcut for :meth:`Interaction.is_user_authorised`.
+
```
Comment (JustaSqu1d) [inline discord/commands/context.py:359]: Missing "There is an alias" note.

**Review by JustaSqu1d — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 6]
```diff
@@ -36,6 +36,8 @@ These changes are available on the `master` branch, but have not yet been releas
   `Permissions.use_external_sounds` and
   `Permissions.view_creator_monetization_analytics`.
   ([#2620](https://github.com/Pycord-Development/pycord/pull/2620))
+- Added helper methods to determine an `Interaction`'s integration type.
```
Comment (JustaSqu1d) [inline CHANGELOG.md:39]:
```suggestion
- Added helper methods to determine the integration type of an `Interaction`.
```

Comment (JustaSqu1d) [reply CHANGELOG.md:39]: ambiguous punctuation

**Review by JustaSqu1d — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 7]
```diff
@@ -36,6 +36,8 @@ These changes are available on the `master` branch, but have not yet been releas
   `Permissions.use_external_sounds`, and
   `Permissions.view_creator_monetization_analytics`.
   ([#2620](https://github.com/Pycord-Development/pycord/pull/2620))
+- Added helper methods to determine the integration type of an `Interaction`.
```
Comment (plun1331) [inline CHANGELOG.md:39]:
```suggestion
- Added helper methods to determine the authorising party of an `Interaction`.
```

**Review by plun1331 — DISMISSED** [review]

File: CHANGELOG.md [hunk 8]
```diff
@@ -36,6 +36,8 @@ These changes are available on the `master` branch, but have not yet been releas
   `Permissions.use_external_sounds`, and
   `Permissions.view_creator_monetization_analytics`.
   ([#2620](https://github.com/Pycord-Development/pycord/pull/2620))
+- Added helper methods to determine the authorising party of an `Interaction`.
```
Comment (JustaSqu1d) [inline CHANGELOG.md:39]:
```suggestion
- Added helper methods to determine the authorizing party of an `Interaction`.
```

**Review by JustaSqu1d — CHANGES_REQUESTED** [review]
For some reason changelog is en-US.

**Review by JustaSqu1d — APPROVED** [review]

**Review by Dorukyum — APPROVED** [review]

**Review by Lulalaby — APPROVED** [review]
we really should stop doing american / british english

**Pull request merged**